### PR TITLE
loadingpage

### PR DESCRIPTION
**Pull request by IsaacBoSim**
https://github.com/Sustainabli/sustainabli/assets/138711892/19016d9b-8a5b-4f95-8b57-4362bbd373b5

### CODE REVIEW

File: src/components/LoadingPage/LoadingPage.css [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/* Layout.css */
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.css:1]: Remove comment

File: src/components/LoadingPage/LoadingPage.css
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.css:1]: Rename file to `LoadingPage.scss`, scss is basically css with OOP properties (i.e. child of parent div also has access to parent's css properties) . Refer to other scss files to see how it's structured. You'll notice that child divs are contained inside a parent div 

Also did you test on various window screens and see how it looks on mobile? (See https://www.sitepoint.com/how-to-simulate-mobile-devices-with-device-mode-in-chrome/ for instructions on how to emulate a website into mobile mode) 

Also test on a few browsers (chrome, edge, firefox). There might be different behavior depending on the window size and browser

File: src/components/LoadingPage/LoadingPage.css [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+.layout {
+  display: flex;
+  height: 100vh;
+}
+
+.nav-sidebar {
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.css:8]: Remove .nav-sidebar if we aren't using it

File: src/components/LoadingPage/LoadingPage.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import './LoadingPage.css';
+
+function LoadingPage() {
+  return (
+    <div className="layout">
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.js:6]: Use React bootstrap container class https://react-bootstrap.netlify.app/docs/layout/grid/. You can probably use the `fluid` property. See examples in the rest of the codebase

File: src/components/LoadingPage/LoadingPage.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      <div className="main-content">
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.js:7]: We probably don't need all of these wrapper divs since we aren't using navsidebar atm

Probably just a single parent container div and then add the 4 square divs (the square divs can be `<divs>`)

File: src/components/LoadingPage/LoadingPage.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        <div className="loading-page">
+          <div className="square-container">
+            <div className="square"></div>
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.js:10]: Can simplify this to `<div className="square"/>`

File: src/components/LoadingPage/LoadingPage.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            <div className="square"></div>
+            <div className="square"></div>
+            <div className="square"></div>
+            <div className="center-square"></div>
+          </div>
+          <p>Loading...</p>
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.js:16]: Don't need `<p>` tags 
We can probably separate the sections into rows/columns (see https://react-bootstrap.netlify.app/docs/layout/grid/ )

File: src/index.js [hunk 7]
```diff
@@ -1,21 +1,32 @@
-import { Suspense } from 'react';
+import React, { Suspense } from 'react';
```
Comment (michaelli00) [inline src/index.js:1]: I don't think we use `React` import here. Remove?

File: src/index.js [hunk 8]
```diff
@@ -1,21 +1,32 @@
...
+         <ProSidebarProvider>
+           <App />
+         </ProSidebarProvider>
+       </RecoilRoot>,
+    </Suspense>,
+
+ document.getElementById('root')
+);
+
+/*ReactDOM.render(
```
Comment (michaelli00) [inline src/index.js:25]: Remove commented code

**Review by michaelli00 — CHANGES_REQUESTED** [review]
Can you update the commit message to be more formal. Format like

```
Create Loading Page

Ticket: https://www.notion.so/Software-001-Loading-Screen-UI-84c650ebefbc4ef09c007ece2788e8f0?pvs=4 

Added bounce/pulse loading effect on the loading page
```

Basic format:
- Title: General short summary of what you did
- Linked ticket
- Paragraph explaining what you did

File: src/components/LoadingPage/LoadingPage.scss [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  background-color: rgba(255, 255, 255, 0.8);
+  height: 100vh;
+}
+
+.square-container {
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.scss:11]: Move square-container within loading-page and square/center-square within square-container. Should look like

```
.loading-page {
  ...
  .square-container {
    ... 
    .square { 

    }
    .center-square { }
  }
}

File: src/components/LoadingPage/LoadingPage.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import './LoadingPage.scss';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+function LoadingPage() {
+  return (
+    <div className="container-fluid loading-page">
```
Comment (michaelli00) [inline src/components/LoadingPage/LoadingPage.js:7]: Use react bootstrap container class https://react-bootstrap.netlify.app/docs/layout/grid/
```
import Container from 'react-bootstrap/Container';
```

File: src/index.js [hunk 11]
```diff
@@ -2,20 +2,22 @@ import { Suspense } from 'react';
...
-  </Suspense>,
-  document.getElementById('root')
+    <Suspense fallback={<LoadingPage/>}>
+       <RecoilRoot>
+         <ProSidebarProvider>
+           <App />
+         </ProSidebarProvider>
+       </RecoilRoot>,
+    </Suspense>,
+
```
Comment (michaelli00) [inline src/index.js:21]: remove extra line

**Review by michaelli00 — APPROVED** [review]
